joblib/test/test_numpy_pickle.py

```diff
@@ -311,17 +307,16 @@ def test_cache_size_warning(tmpdir):
     a = rnd.random_sample((10, 2))
 
     for cache_size in (None, 0, 10):
-        with warnings.catch_warnings(record=True) as caught_warnings:
-            warnings.simplefilter("always")
+        with warns(None) as warninfo:
             numpy_pickle.dump(a, filename, cache_size=cache_size)
-            expected_nb_warnings = 1 if cache_size is not None else 0
-            assert len(caught_warnings) == expected_nb_warnings
-            for warn in caught_warnings:
-                assert warn.category == DeprecationWarning
-                assert (warn.message.args[0] ==
-                        "Please do not set 'cache_size' in joblib.dump, this "
-                        "parameter has no effect and will be removed. You "
-                        "used 'cache_size={0}'".format(cache_size))
+        expected_nb_warnings = 1 if cache_size is not None else 0
+        assert len(warninfo) == expected_nb_warnings
+        for w in warninfo:
```

Comment:
This loop over 0 or 1 element is a bit misleading. Oh well I'll tackle that myself.